runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2314,7 +2400,21 @@ bool TR_EscapeAnalysis::checkOverlappingLoopAllocation(TR::Node *node, TR::Node
 
    _visitedNodes->set(node->getGlobalIndex());
 
-   if ((_valueNumberInfo->getValueNumber(node) == _valueNumberInfo->getValueNumber(useNode)) && (node != allocNode))
+   if (node->getOpCode().isStore() && node->getSymbol()->isAutoOrParm())
+      {
+      if (node->getFirstChild() == allocNode)
+         _aliasesOfAllocNode->set(node->getSymbolReference()->getReferenceNumber());
+      else if (!_visitedNodes->get(node->getFirstChild()->getGlobalIndex()) && node->getFirstChild()->getOpCode().isLoadVarDirect() && node->getFirstChild()->getSymbol()->isAutoOrParm() &&
+               _aliasesOfAllocNode->get(node->getFirstChild()->getSymbolReference()->getReferenceNumber()))
+         _aliasesOfAllocNode->set(node->getSymbolReference()->getReferenceNumber());
+      else
+         _aliasesOfAllocNode->reset(node->getSymbolReference()->getReferenceNumber());
+      }
+
+   if ((_valueNumberInfo->getValueNumber(node) == _valueNumberInfo->getValueNumber(useNode)) &&
```

Comment:
for efficiency perhaps do the node != allocNode first since it is cheaper?

Reply:
Cleaned this up in commit 414b74b10957a8ae1b8a4426f0c29dec2a905275